src/connectivity/mod.rs

```diff
@@ -736,6 +737,134 @@ pub fn graph_all_pairs_all_simple_paths(
     ))
 }
 
+/// Return a longest simple path in the graph
+///
+/// This function searches computes all pairs of all simple paths and returns
+/// a path of the longest length from that set. It is roughly equivalent to
+/// running something like::
+///
+///     from rustworkx import all_pairs_all_simple_paths
+///
+///     max((y.values for y in all_pairs_all_simple_paths(graph).values()), key=lambda x: len(x))
```

Comment:
@mtreinish, thank you for implementing this function for me. After running multiple experiments from within Qiskit, I believe that `longest_simple_path` works correctly. However, running the documented equivalent formulation raises an error. First, I believe it should be `y.values()` and not `y.values`, but this is still not fully correct. The following does seem to work:
```
simple_path_pairs = rx.all_pairs_all_simple_paths(graph)
longest_path = max((u for y in simple_path_pairs.values() for z in y.values() for u in z), key=lambda x: len(x))
```